backend/union/union.go

```diff
@@ -616,7 +637,18 @@ func (f *Fs) ListR(ctx context.Context, dir string, callback fs.ListRCallback) (
 			entriess = append(entriess, uEntries)
 			mutex.Unlock()
 			return nil
-		})
+		}
+		do := u.Features().ListR
+		if do != nil {
+			err = do(ctx, dir, callback)
+		} else {
+			// if we're here, we're gonna do a standard recursive traversal
+			list := walk.NewListRHelper(callback)
```

Comment:
Why not this? Saves having to have your `recurse` function?

```go
err := walk.ListR(ctx, u, dir, true, -1, walk.ListAll, callback)
```

Reply:
My fault. I did not see this function and just copy implementation from cache.